Docker Desktop 4.8.0 Release notes (#14630)
docker/docker.github.io: desktop/mac/release-notes/index.md

Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/projects/1){: target="_blank" rel="noopener" class="_"} to see what's coming next.

## Docker Desktop 4.8.0

> 2022-05-05

Reviewer: ```suggestion
2022-05-06
```

> chip](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){:
> .button .primary-btn }

### New

> - Released Docker Desktop for Linux, supporting deb and rpm packaging formats.

Reviewer 2: I don't think we need to say "supporting deb and rpm packaging formats"

- Beta release of Docker Extensions and Extensions SDK.

> - [Compose V2 has been marked GA](https://www.docker.com/blog/announcing-compose-v2-general-availability/) 

Reviewer 3: ```suggestion
- New Docker Homepage: Run popular images and discover how to use them.
- [Compose V2 has been marked GA](https://www.docker.com/blog/announcing-compose-v2-general-availability/) 
```

> - [Compose V2 has been marked GA](https://www.docker.com/blog/announcing-compose-v2-general-availability/) 

Reviewer 2: "has now reached GA"

### Upgrades
- [Go 1.18.1](https://golang.org/doc/go1.18)

> ### Bug fixes and minor changes

Reviewer 2: We need for-mac bug numbers on all of these, where they exist.

Author: Done! Only 2 is what I've found.

> chip](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){:
> .button .primary-btn }

### New

> - Released Docker Desktop for Linux.

Reviewer 4: Could we have a link to the Linux docs?

Author: Sure, is this link ok? https://docs.docker.com/desktop/linux/

Reviewer 4: Yes, thanks @lorenrh

> - Beta release of Docker Extensions and Extensions SDK.

Reviewer: Can we add a link to the Extensions doc - https://docs.docker.com/desktop/extensions

> - New Docker Homepage: Run popular images and discover how to use them.

Reviewer 4: ```suggestion
- Created a Docker Homepage where you can run popular images and discover how to use them.
```

> - [Compose V2 has now reached GA](https://www.docker.com/blog/announcing-compose-v2-general-availability/)

Reviewer 4: ```suggestion
- [Compose V2 is now GA](https://www.docker.com/blog/announcing-compose-v2-general-availability/)
```

### Upgrades

- [Compose v2.5.0](https://github.com/docker/compose/releases/tag/v2.5.0)
- [Go 1.18.1](https://golang.org/doc/go1.18)

### Bug fixes and minor changes

> - Docker Desktop now uses the operating system proxy settings whe available on MacOS and Windows.

Reviewer 4: ```suggestion
- Docker Desktop now uses the operating system's proxy settings where available, on MacOS and Windows.
```

- Fixed a bug that showed Remote Repositories in the Dashboard when running behind a proxy.

> - Fix vpnkit establishing and blocking client connection even if the server is gone. See [docker/for-mac#6235](https://github.com/docker/for-mac/issues/6235).

Reviewer 4: ```suggestion
- Fixed vpnkit establishing and blocking the client connection even if the server is gone. See [docker/for-mac#6235](https://github.com/docker/for-mac/issues/6235).
```

> - Improvements on volume tab in the Docker Desktop Dashboard:

Reviewer 4: ```suggestion
- Made improvements to the Volume tab in Docker Desktop:
```

>   - Show volume size.

Reviewer 4: ```suggestion
  - Volume is displayed.
```

  - Columns can be resized, hidden and reordered.

>   - Columns sort order and hidden state is persisted, even after Docker Desktop restarts.

Reviewer 4: ```suggestion
  - A columns sort order and hidden state is persisted, even after Docker Desktop restarts.
```

  - Row selection is persisted when switching between tabs, even after Docker Desktop restarts.

> - Fixed a bug in the Dev Environments tab that did not add a scroll when more items were added in the screen.

Reviewer 4: ```suggestion
- Fixed a bug in the Dev Environments tab that did not add a scroll when more items were added to the screen.
```

> - Standardise the header title and action in the dashboard screens.

Reviewer 4: ```suggestion
- Standardised the header title and action in the Dashboard.
```

> - Add support for downloading Registry Access Management policies through HTTP proxies.

Reviewer 4: ```suggestion
- Added support for downloading Registry Access Management policies through HTTP proxies.
```

> - Add support for downloading Registry Access Management policies through HTTP proxies.

Reviewer: @christophermclellan Should we add this note given that RAM is in private beta?

> - Fixed an issue related to empty remote repositories when the machine went on sleep for long time.

Reviewer 4: ```suggestion
- Fixed an issue related to empty remote repositories when the machine was in sleep mode for an extended period of time.
```

> - Fixed a bug were dangling images were not selected in the cleanup process if their name was not marked as "<none>" but their tag is.

Reviewer 4: ```suggestion
- Fixed a bug where dangling images are not selected in the cleanup process if their name is not marked as "<none>" but their tag is.
```

> - Docker Desktop's icon now matches Big Sur Style guide. See [docker/for-mac#6235](https://github.com/docker/for-mac/issues/5536).

Reviewer 5: The issue number shown in the link is wrong. And we now have a second PR merged addressing another Dock icon issue, that's why I added it here.

```suggestion
- Docker Desktop's icon now matches Big Sur Style guide. See [docker/for-mac#5536](https://github.com/docker/for-mac/issues/5536).
- [mac] Fixed a problem with duplicate Dock icons and Dock icon not working as expected. Fixes [docker/for-mac#6189](https://github.com/docker/for-mac/issues/6189).
```

> - Improve error message when `docker pull` fails because an HTTP proxy is required.

Reviewer 4: ```suggestion
- Improved the error message when `docker pull` fails because an HTTP proxy is required.
```

> - On the Dashboard, added clear on all search inputs.

Reviewer 4: Added the ability to clear the search
```suggestion
- Added the ability to clear the search bar easily in Docker Desktop. 
```

- Renamed the "Containers / Apps" tab to "Containers".

> - Introduced reading system proxy, users no longer need to manually configure proxies unless it differs from their OS level proxy.

Reviewer 4: ```suggestion
- Introduced reading system proxy. You no longer need to manually configure proxies unless it differs from your OS level proxy.
```

> - Better support for `Cmd+Q` shortcut.

Reviewer 4: ```suggestion
- Improved support for the `Cmd+Q` shortcut.
```

> 

Reviewer 5: Please add
```
- [mac] Fixed a problem with duplicate Dock icons and Dock icon not working as expected. Fixes [docker/for-mac#6189](https://github.com/docker/for-mac/issues/6189).
```

Author: Will do!

> chip](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){:
> .button .primary-btn }

### New

> - Released Docker [Desktop for Linux](https://docs.docker.com/desktop/linux/).

Reviewer 2: ```suggestion
- Released [Docker Desktop for Linux](https://docs.docker.com/desktop/linux/).
```

- Beta release of [Docker Extensions](https://docs.docker.com/desktop/extensions) and Extensions SDK.
- Created a Docker Homepage where you can run popular images and discover how to use them.
- [Compose V2 is now GA](https://www.docker.com/blog/announcing-compose-v2-general-availability/)

### Upgrades

- [Compose v2.5.0](https://github.com/docker/compose/releases/tag/v2.5.0)
- [Go 1.18.1](https://golang.org/doc/go1.18)

> 

Reviewer 2: Kube 1.24

  - Row selection is persisted when switching between tabs, even after Docker Desktop restarts.
- Fixed a bug in the Dev Environments tab that did not add a scroll when more items were added to the screen.
- Standardised the header title and action in the Dashboard.
- Added support for downloading Registry Access Management policies through HTTP proxies.
- Fixed an issue related to empty remote repositories when the machine is in sleep mode for an extended period of time.
- Fixed a bug where dangling images were not selected in the cleanup process if their name was not marked as "<none>" but their tag is.
- Docker Desktop's icon now matches Big Sur Style guide. See [docker/for-mac#5536](https://github.com/docker/for-mac/issues/5536).
- Fixed a problem with duplicate Dock icons and Dock icon not working as expected. Fixes [docker/for-mac#6189](https://github.com/docker/for-mac/issues/6189).
- Improved the error message when `docker pull` fails because an HTTP proxy is required.
- Added the ability to clear the search bar easily in Docker Desktop. 
- Renamed the "Containers / Apps" tab to "Containers".

> - Introduced reading system proxy. You no longer need to manually configure proxies unless it differs from your OS level proxy.

Reviewer 2: This is a dupe of line 49.

Author: I'll keep this release note instead of line 49. It's more understandable IMO.
